### ISSUE TITLE
Informix DDL statements generate invalid SQL for timestamp columns

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14313]
A new integration test query fails on Informix:

```sql
CREATE TABLE T_COMP_CLIENT_SIDE_DEP (
  ID INTEGER NOT NULL,
  -- ...
  X_MODIFIED_AT DATETIME,
  PRIMARY KEY (ID) CONSTRAINT PK_T_COMP_CLIENT_SIDE_DEP
);
```

We should use the cast type for `DATETIME`, e.g. `DATETIME YEAR TO FRACTION (5)`

**Comment by lukaseder** [comment]
This is best solved via https://github.com/jOOQ/jOOQ/issues/10376 in jOOQ 3.18. But the backport will use a simple `if` check for this specific case.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.6 (https://github.com/jOOQ/jOOQ/issues/14314), and 3.16.12 (https://github.com/jOOQ/jOOQ/issues/14315)

**Comment by lukaseder** [comment]
See also related follow up issue: https://github.com/jOOQ/jOOQ/issues/14316
